strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/storage/validation/groupId/MavenGroupIdLowercaseValidatorTest.java

```diff
@@ -0,0 +1,58 @@
+package org.carlspring.strongbox.storage.validation.groupId;
+
+import org.carlspring.strongbox.artifact.coordinates.MavenArtifactCoordinates;
+import org.carlspring.strongbox.providers.io.RepositoryFileAttributes;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.storage.validation.exceptions.LowercaseValidationException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Spy;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+
+/**
+ * Created by dinesh on 12/7/17.
+ */
+
+public class MavenGroupIdLowercaseValidatorTest
+{
+
+    @Spy
```

Comment:
For my personal understanding, what does the `@Spy` annotation do? (I haven't used this myself before).

Reply:
Spy is used for partial mocking.